app/api/v2/managers/operation_api_manager.py

```diff
@@ -20,3 +49,51 @@ async def get_operation_object(self, operation_id: str, access: dict):
         if operation.match(access):
             return operation
         raise JsonHttpForbidden(f'Cannot view operation due to insufficient permissions: {operation_id}')
+
+    async def setup_operation(self, data: dict, access: BaseWorld.Access):
+        """Applies default settings to an operation if data is missing."""
+        planner_id = data.pop('planner', {}).get('id', '')
+        data['planner'] = await self._construct_and_dump_planner(planner_id)
+        adversary_id = data.pop('adversary', {}).get('adversary_id', '')
+        data['adversary'] = await self._construct_and_dump_adversary(adversary_id)
+        fact_source_id = data.pop('source', {}).get('id', '')
+        data['source'] = await self._construct_and_dump_source(fact_source_id)
+        operation = OperationSchema().load(data)
+        await operation.update_operation_agents(self.services)
+        allowed = self._get_allowed_from_access(access)
+        operation.set_operation_access(allowed)
+        operation.set_start_details()
+        return operation
+
+    async def _construct_and_dump_planner(self, planner_id: str):
+        planner = await self.services['data_svc'].locate('planners', match=dict(id=planner_id))
```

Comment:
This should be `planner_id` instead of `id`. Throws an error for me if it's just `id`